user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/3-sqs.md

```diff
@@ -0,0 +1,309 @@
+
+### Generate the lambda handler
+
+With SQS contracts, you can generate a natively typed lambda handler which
+
+- **validates** the messages against the contract to unsure the message body and message attributes format
+- **parse** the message body and message attributes to handle proper js object.
+- abstract the iteration on the **batched messages**
+- abstract the return of **failed messages** to SQS
+
+Simply write:
+
+```ts
+import { getHandler } from '@swarmion/serverless-contracts';
+import { ajv } from 'libs/ajv';
+
+export const main = getHandler(mySqsContract, { ajv })(async message => {
+  const { toto } = message.body; // parsed and typed with the correct keys
+
+  // write your business logic for one message
+});
+```
+
+:::info
+If one of the message handler fails, **the global handler will catch the error** and return the id of the failed messages to SQS. The return of the handler will be like:
+
+```json
+{
+  "batchItemFailures": [
+    {
+      "itemIdentifier": "messageId"
+    }
+  ]
+}
+```
+
+:::
+
+:::info
+The default body parser is `JSON.parse`. You can provide your own body parser with the `bodyParser` option.
+Use explicit `undefined` to disable body parsing.
+
+```ts
+import { getHandler } from '@swarmion/serverless-contracts';
+
+const handler = getHandler(myContract, {
+  ajv,
+  bodyParser: undefined,
+})(async message => {
+  const { body } = message; // body is a raw string
+  // ...
+});
+```
+
+:::
+
+:::info
+Regarding the `ajv` option, we advise you to use a singleton instance of ajv that you define in a separate file. This way, you can use the same instance for all your contracts and middlewares.
+
+[See an example](../../how-to-guides/migration-guides/ajv-dependency-injection#share-a-singleton-ajv-instance-across-the-whole-project)
+:::
+
+:::caution
+This handler also provides a body validation that will throw an error if there is a mismatch with the `messageBodySchema`. This ensure that invalid messages will not be mistakenly taken into account. However, be sure to set up an invalid message failure flow, for example with a [DLQ](https://www.serverless.com/blog/lambda-destinations/).
+
+If you still wish to disable this behavior, you can use the optional second argument in the `getHandler` feature.
+If you do so, you can omit the `ajv` option.
+
+```ts
+import { getHandler } from '@swarmion/serverless-contracts';
+
+const handler = getHandler(myContract, {
+  validateBody: false,
+})(async message => {
+  // ...
+});
+```
+
+:::
+
+Alternatively, if you wish to handle the batch behavior by yourself, you can set the `handleBatchedRecords` option to `false`.
+
+```ts
+import { getHandler } from '@swarmion/serverless-contracts';
+import { ajv } from 'libs/ajv';
+
+export const main = getHandler(mySqsContract, {
+  ajv,
+  handleBatchedRecords: false,
+})(async ({ Records }) => {
+  Records.forEach(message => {
```

Comment:
Nit: the AWS API convention is a bit weird with its UpperCamelCase object keys, I would prefer to use `records`, WDYT?

Reply:
I agree, I changed it